pkg/providers/version/version.go

```diff
@@ -69,39 +70,39 @@ func NewDefaultProvider(kubernetesInterface kubernetes.Interface, cache *cache.C
 }
 
 func (p *DefaultProvider) Get(ctx context.Context) (string, error) {
-	var version string
-	var versionSource string
-	if version, ok := p.cache.Get(kubernetesVersionCacheKey); ok {
-		return version.(string), nil
+	if version := *p.previousVersion.Clone(); version != "" {
+		return version, nil
 	}
-	output, err := p.eksapi.DescribeCluster(ctx, &eks.DescribeClusterInput{
-		Name: lo.ToPtr(options.FromContext(ctx).ClusterName),
-	})
-	if err != nil {
-		if !awserrors.IsAccessDenied(err) {
-			return "", err
-		}
-		output, err := p.kubernetesInterface.Discovery().ServerVersion()
-		if err != nil {
-			return "", fmt.Errorf("getting kubernetes version from the kubernetes API")
-		} else if output != nil {
-			version = fmt.Sprintf("%s.%s", output.Major, strings.TrimSuffix(output.Minor, "+"))
-			versionSource = "Kubernetes API"
-		}
-	} else if lo.FromPtr(output.Cluster.Version) != "" {
-		version = *output.Cluster.Version
+	return "", fmt.Errorf("kubernetes version not yet cached")
+}
+
+func (p *DefaultProvider) UpdateVersion(ctx context.Context) error {
+	var version, versionSource string
+	var err error
+
+	if options.FromContext(ctx).EKSControlPlane {
+		version, err = p.getEKSVersion(ctx)
```

Comment:
nit: let's handle the immediately, rather than further down in the function. Doing this makes the error handling easier to keep track of generally. Same applies for in the else branch.
```suggestion
		version, err = p.getEKSVersion(ctx)
		if err != nil {
		    return err
		}
```

Reply:
Okay perfect I am also removing the third conditional because that actually is not needed as a fall back anymore.